src/components/UpdateAppModal/BaseUpdateAppModal.js

```diff
@@ -0,0 +1,94 @@
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    TouchableOpacity, Text,
+} from 'react-native';
+import CONST from '../../CONST';
+import HeaderWithCloseButton from '../HeaderWithCloseButton';
+import Modal from '../Modal';
+import styles from '../../styles/styles';
+import themeColors from '../../styles/themes/default';
+
+/**
+ * Modal render prop component that exposes modal launching triggers that can be used
+ * to display a full size image or PDF modally with optional confirmation button.
+ */
+
+const propTypes = {
+    // Optional callback to fire when we want to trigger the update.
+    onConfirm: PropTypes.func,
+
+    // Version string for the app to update to.
+    // eslint-disable-next-line react/no-unused-prop-types
+    version: PropTypes.string,
+};
+
+const defaultProps = {
+    onConfirm: null,
+    version: '',
+};
+
+class BaseUpdateAppModal extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isModalOpen: true,
+        };
+
+        this.submitAndClose = this.submitAndClose.bind(this);
+    }
+
+    /**
+     * Execute the onConfirm callback and close the modal.
+     */
+    submitAndClose() {
+        this.props.onConfirm(this.state.file);
+        this.setState({isModalOpen: false});
+    }
+
+    render() {
+        return (
+            <>
+                <Modal
+                    type={CONST.MODAL.MODAL_TYPE.RIGHT_DOCKED}
```

Comment:
Up to you, but you should see what happens when you leave this `type` prop out entirely. That will cause the modal to be centered in the screen rather than right-docked, which to me feels maybe more appropriate for this modal.